airflow/providers/google/cloud/transfers/gdrive_to_gcs.py

```diff
@@ -0,0 +1,134 @@
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.google.cloud.hooks.gcs import GCSHook
+from airflow.providers.google.suite.hooks.drive import GoogleDriveHook
+from airflow.utils.decorators import apply_defaults
+
+
+class GoogleDriveToGCSOperator(BaseOperator):
+    """
+    Writes a Google Drive file into Google Cloud Storage.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GoogleDriveToGCSOperator` FIXME
+
+    :param destination_bucket: The destination Google cloud storage bucket where the
+        report should be written to. (templated)
+    :param destination_bucket: str
+    :param destination_object: The Google Cloud Storage object name for the object created by the operator.
+        For example: ``path/to/my/file/file.txt``.
+    :type destination_object: str
+    :param folder_id: The folder id of the folder in whhich the Google Drive file resides
+    :type folder_id: str
+    :param file_name: The name of the file residing in Google Drive
+    :type file_name: str
+    :param gcp_conn_id: The GCP connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :type impersonation_chain: Union[str, Sequence[str]]
+    """
+
+    template_fields = [
+        "destination_bucket",
+        "destination_object",
+        "folder_id",
+        "file_name",
+        "impersonation_chain",
+    ]
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        destination_bucket: str,
+        destination_object: str,
+        folder_id: Optional[str] = None,
+        file_name: Optional[str] = None,
+        gcp_conn_id: str = "google_cloud_default",
+        delegate_to: Optional[str] = None,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ) -> None:
+        if folder_id and not file_name:
+            raise AirflowException("Both folder_id and file_name must be set. Received only folder_id")
```

Comment:
Why folder_id and file_name have to be set? Could be possible to pass just file id? (`providers.google.suite.hooks.drive.GoogleDriveHook.upload_file` returns file id)

Reply:
After some thought, I think it's less practical to use a `file_id` for the operator because it would be very clumsy to pass the `file_id` to the task. I don't think there are any clear advantages of passing a file id instead of a `file_name` and `folder_id` combination, but the file name and folder id combination is dynamic and wouldn't require the user to first discover the file id.

I believe that this can only function in a folder, because service accounts can't access files that aren't in shared folders.

Reply:
I'll try to verify that service accounts can't access shared drives...

Reply:
I might've overlooked the `driveId` param in the files.list method. I don't have a Google Workspace account, so I can't test this without access to a shared drive. Do you know how I can get access to one?

I think this would only slightly change the implementation -- `folder_id` would be optional, the files.list `query` param would be dynamic based on the presence or absence of `folder_id`, and `drive_id` would be an optional init param.

Reply:
I created a trial Google Workspace account to figure out the functionality. I plan to implement the strategy in my previous comment: 
1. Retain `folder_id` as a required parameter to remove any ambiguity or confusion
2. Add `drive_id` as an optional parameter to allow users to search Shared Drives.

**NOTE:** It is possible to search for a file at the root of a shared drive. The user would just have to pass the shared drive's Id to the `drive_id` and `folder_id` parameters. I'll be sure to call this out in the documentation.